src/javascript/tests/booleans.rs

```diff
@@ -0,0 +1,41 @@
+use crate::assert_js;
+
+#[test]
+fn boolean_literals() {
+    assert_js!(
+        r#"
+fn go() {
+    True
+    False
+    Nil
+}
+"#,
+        r#""use strict";
+
+function go() {
+  true;
+  false;
+  return null;
```

Comment:
I think `undefined` might be a closer fit as it's what is returned by JavaScript non-returning expressions.

Reply:
I think semantically `null` is closer. undefined is meant to be more due to JS's ability to declare a variable before using it.
Personally I often handwrite JS (or typescript) with null so that I know undefined means I've fogotten an execution path.

Tbh, I suspect there will a large number of developers who could argue either way.

Reply:
I think `null` is typically used to say "this field exists but it is empty". i.e. `{ name: null }`. This is how we use `None`.

`undefined` is used when something returns nothing (`let x = non_returning_function()`) or when accessing something that does not exist `{ "ok": 1 }.missing_field`, which is how we use `Nil`.

Reply:
> `undefined` is used when something returns nothing

Not in my experience. In fact in typescript I see a lot of deliberate use of `null` so that you use the checker to easily find bugs where you have forgotten a return or to handle the fallback case.

As mentioned not a hill I want to die on. I can disagree and commit. And if you'd rather have undefined, I'll make it so.

Reply:
I think we can avoid the need for the runtime error handling behaviour through the type system, it'll never let you return `Nil` erroneously. Let's go with `undefined`.

Reply:
Done